datadog/resource_datadog_integration_aws.go

```diff
@@ -175,6 +183,11 @@ func buildDatadogIntegrationAwsStruct(d *schema.ResourceData) *datadogV1.AWSAcco
 		iaws.SetCspmResourceCollectionEnabled(vBool)
 	}
 
+	if v, ok := d.GetOk("extended_resource_collection_enabled"); ok && v.(string) != "" || iaws.GetCspmResourceCollectionEnabled() {
+		vBool, _ := strconv.ParseBool(v.(string))
+		iaws.SetExtendedResourceCollectionEnabled(vBool || iaws.GetCspmResourceCollectionEnabled())
```

Comment:
Not sure I'm reading this right, but this seems weird to me - does this mean if the customer sets extended=false, and cspm=true the request will succeed as the terraform provider will send extended=true and cspm=true to our api? I feel like in that case the request should fail since the parameters provided were invalid, rather than fixing the parameters.

Reply:
Looks like I messed up while attempting to factor out some nested if-statements into boolean ops. I’ll fix it up to fail if extended is explicitly false when cspm is true.